### Zopper Device Insurance Attach Analysis :- Data Understanding

In [1]:
import pandas as pd
import os
df_raw = pd.read_excel(r"C:\Users\giris\OneDrive\Documents\Jumbo_and_company_attach%.xlsx")

In [2]:
df_raw.head()

Branch               Store_Name   Dec   Nov   Oct   Sep   Aug
0  Delhi_Ncr      Delhi(Janakpuri) Br  0.23  0.17  0.16  0.25  0.24
1  Delhi_Ncr      Haryana(Gurgaon) Br  0.21  0.26  0.15  0.28  0.04
2  Delhi_Ncr     Up(Greater Noida) Br  0.25  0.36  0.30  0.41  0.43
3       Pune         Pune(Bhosari) Br  0.33  0.33  0.36  0.13  0.32
4    Gujarat  Ahmedabad(Maninagar) Br  0.19  0.11  0.14  0.21  0.17

**Structural Checks of the Data**

In [3]:
df_raw.shape
df_raw.info()
df_raw.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 163 entries, 0 to 162
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Branch      163 non-null    object 
 1   Store_Name  163 non-null    object 
 2   Dec         163 non-null    float64
 3   Nov         163 non-null    float64
 4   Oct         163 non-null    float64
 5   Sep         163 non-null    float64
 6   Aug         163 non-null    float64
dtypes: float64(5), object(2)
memory usage: 9.0+ KB


Dec         Nov         Oct         Sep         Aug
count  163.000000  163.000000  163.000000  163.000000  163.000000
mean     0.217239    0.217117    0.170920    0.167301    0.128589
std      0.173270    0.131246    0.116125    0.134518    0.116640
min      0.000000    0.000000    0.000000    0.000000    0.000000
25%      0.095000    0.130000    0.100000    0.080000    0.035000
50%      0.200000    0.200000    0.160000    0.150000    0.110000
75%      0.300000    0.295000    0.240000    0.245000    0.190000
max      1.000000    0.700000    0.710000    0.800000    0.600000

In [4]:
# Number of branches
df_raw['Branch'].nunique()

6

In [5]:
# Number of stores
df_raw['Store_Name'].nunique()


163

**Conclusion**

The dataset contains complete store-level device insurance attach percentage data for 163 stores across multiple branches from August to December, with no missing values and all monthly attach rates available in numeric format. The average attach rate shows a clear upward trend from August (12.9%) to November–December (~21.7%), indicating improved performance toward the year end. However, the wide spread in attach percentages, ranging from 0% to as high as 100%, along with high variability across stores, suggests that performance is driven largely by store-level execution rather than uniform customer demand, highlighting significant opportunity for targeted operational and training interventions.